rascal2/ui/presenter.py

```diff
@@ -23,8 +31,61 @@ def createProject(self, name: str, save_path: str):
             The name of the project.
         save_path : str
             The save path of the project.
+
         """
 
         self.model.createProject(name, save_path)
         # TODO if the view's central widget is the startup one then setup MDI else reset the widgets.
         self.view.setupMDI()
+
+    def getControlsAttribute(self, setting) -> Any:
+        """Get the value of an attribute in the model's Controls object.
+
+        Parameters
+        ----------
+        setting : str
+            Which setting in the Controls object should be read.
+
+        Returns
+        -------
+        value : Any
+            The value of the setting in the model's Controls object.
+        """
+        value = getattr(self.model.controls, setting)
+        return value
+
+    def editControls(self, setting: str, value: Any) -> bool:
+        """Edit a setting in the Controls object.
+
+        Parameters
+        ----------
+        setting : str
+            Which setting in the Controls object should be changed.
+        value : Any
+            The value which the setting should be changed to.
+
+        Returns
+        -------
+        bool
+            Whether the edit was successful.
+        """
+        # FIXME: without proper logging,
+        # we have to check validation in advance because PyQt doesn't return
+        # the exception, it just falls over in C++
+        # also doing it this way stops bad changes being pushed onto the stack
+        # also suppress warnings (we get warning for setting params not matching
+        # procedure on initialisation) to avoid clogging stdout
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            try:
+                self.model.controls.model_validate({setting: value})
+            except ValidationError as err:
+                self.last_validation_error = err
```

Comment:
I think you could return the validation error instead of a bool then avoid storing  `self.last_validation_error` in presenter i.e.
 
```
 with warnings.catch_warnings():
      warnings.simplefilter("ignore")
      try:
          self.model.controls.model_validate({setting: value})
      except ValidationError as err:
          return  err
     self.view.undo_stack.push(commands.editControls(self.model.controls, setting, value))
```

Reply:
half-implementing a 'result type' doesn't feel very Pythonic - even simpler, i'll just let `model_validate` raise the error and then handle it on the other side